coalib/processes/Processing.py

```diff
@@ -329,26 +324,32 @@ def instantiate_bears(section,
     :param console_printer:  Object to print messages on the console.
     :return:                 The local and global bear instance lists.
     """
-    local_bear_list = [bear
-                       for bear in filter_raising_callables(
-                           local_bear_list,
-                           RuntimeError,
-                           section,
-                           message_queue,
-                           timeout=0.1,
-                           debug=debug)]
-
-    global_bear_list = [bear
-                        for bear in filter_raising_callables(
-                            global_bear_list,
-                            RuntimeError,
-                            file_dict,
-                            section,
-                            message_queue,
-                            timeout=0.1,
-                            debug=debug)]
-
-    return local_bear_list, global_bear_list
+    instantiated_local_bear_list = []
+    instantiated_global_bear_list = []
+    for bear in local_bear_list:
+        try:
+            debugger = debug_flag_checker(bear, debug_bears=debug_bears)
+            instantiated_local_bear_list.append(bear(section,
+                                                     message_queue,
+                                                     timeout=0.1,
+                                                     debugger=debugger))
+        except RuntimeError:
+            if debug:
+                raise
```

Comment:
No not `re-raising` the `RuntimeError` it will be raised only when `debug` will set `True`. (If there is any RuntimeError then only)
By default `debug` is `False`. If user passed the arg `--debug` then this `debug` will set `True` and after that, it will catch the `RuntimeError` and raise the `RuntimeError` and `ipdb` will gets raised. If I removed the `debug == True` and let the exception gets `rasied` directly, then it will break the coala every time if any bear raises `RuntimeError` during its initialization, and why `RuntimeError` as the same process was happening on previously also.

And don't want to remove `debug==True` because here I like the concept of @userzimmermann  to raise a `ipdb` only when `--debug` is getting called.

Reply:
Ah understood. Could you please write a comment there then? That makes it clear for others as well :)

(sorry for responding so late, missed this comment)

Reply:
Also for the global bear. Even if it's duplicated, people see it immediately :+1: